### DISCUSSION TITLE
Unable to run at WSL2 (with Chrome)

### DISCUSSION THREAD

**Discussion by valentijnscholten** [getgauge/taiko#1761] [General]
Hi,

I'm doing most of my development on WSL2, so would like to do the same for writing tests in taiko. Running Ubuntu20.04 on WSL2.

`sudo npm install -g taiko`

succeeds

`taiko`

succeeds

`openBrowser()`

fails:

```
> openBrowser()
 ✘ Error: Chromium revision is not downloaded. Provide TAIKO_BROWSER_PATH or Install taiko again to download bundled chromium., run `.trace` for more info.
```

So it doesn't seem to install chromium. Anyway, I'd like to use chrome which is installed in the windows host system.

```
export TAIKO_BROWSER_PATH=/mnt/c/Program\ Files\ \(x86\)/Google/Chrome/Application/chrome.exe
taiko
openBrowser()
```

This opens chrome on windows. Shows some warnings about userprofile (which might be OK).
But taiko cannot connect after starting:

```
 ✘ Error: connect ECONNREFUSED 127.0.0.1:37908, run `.trace` for more info.
````

I tried making host and port explicit. host.docker.internal is the hostname for the windows host, which resolves OK inside WSL and on Windows itself. I opened port 12345 in the windows firewall.

```
> openBrowser({headless: false, host:'host.docker.internal', port: 12345, args: ['--remote-debugging-address=host.docker.internal', '--remote-debugging-port=12345']})
```

Nothing happens. Taiko just seems to hang (which may indicate a network issue conntecting to the host). Same for `headless: true`.

Anybody able to get it to work with WSL2?

**Comment by valentijnscholten** [comment]
In the mean time I got Taiko running under WSL2 in combination with an X Server, the usual way to run GUI applications from WSL2. Still would be nice if all that is not needed and we could use the chrome installed on the windows host.

**Comment by flogr** [comment]
Can anyone say if it works on WSL2 under Windows 11? One of the new features of the upgraded WSL should be "native" support of GUI applications, right?
